### PR TITLE
Improve prospector config

### LINKED ISSUE #1033: Prospector config improvement
# Prospector configuration defaults

The current filebeat prospecto configuration is somehow confusing as people forget the - for the prospectors. To make it easier to understand and less error prone, the intention is to place one configuration directly behind the -. Below are 2 options, I suggest to go with the input_type option.

## Current version

```
filebeat:
  # List of prospectors to fetch data.
  prospectors:
    # Each - is a prospector. Below are the prospector specific configurations
    -
      # Paths that should be crawled and fetched. Glob based paths.
      # To fetch all ".log" files from a specific level of subdirectories
      # /var/log/*/*.log can be used.
      # For each file found under this path, a harvester is started.
      # Make sure not file is defined twice as this can lead to unexpected behaviour.
      paths:
        - /var/log/*.log

      # Type of the files. Based on this the way the file is read is decided.
      # The different types cannot be mixed in one prospector
      #
      # Possible options are:
      # * log: Reads every line of the log file (default)
      # * stdin: Reads the standard in
      input_type: log


    # Additional prospector
    #-
      # Configuration to use stdin input
      #input_type: stdin
```

## input_type option

This is my preferred option, as input_type must exist for each prospector. The default prospector is log and it would also work without it, but I think it is good to get people to have it in.

The part I don't like yet is the way the documentation looks. It would be nice to get some ideas here.

```
filebeat:
  # List of prospectors to fetch data.
  prospectors:
    # Each - is a prospector. Below are the prospector specific configurations
	
      # Type of the files. Based on this the way the file is read is decided.
      # The different types cannot be mixed in one prospector
      #
      # Possible options are:
      # * log: Reads every line of the log file (default)
      # * stdin: Reads the standard in
    - input_type: log
	
      # Paths that should be crawled and fetched. Glob based paths.
      # To fetch all ".log" files from a specific level of subdirectories
      # /var/log/*/*.log can be used.
      # For each file found under this path, a harvester is started.
      # Make sure not file is defined twice as this can lead to unexpected behaviour.
      paths:
        - /var/log/*.log


    # Additional prospector

	   # Configuration to use stdin input
    #- input_type: stdin
      
```

## paths option

As most people use the log type and paths are required, this one makes it obvious how to define paths.

Concerning comments it has the same issues as the input_type option. 

For a different input_type either input_type could be put on a new line or we mix the different options (as below).

```
filebeat:
  # List of prospectors to fetch data.
  prospectors:
    # Each - is a prospector. Below are the prospector specific configurations
    
      # Paths that should be crawled and fetched. Glob based paths.
      # To fetch all ".log" files from a specific level of subdirectories
      # /var/log/*/*.log can be used.
      # For each file found under this path, a harvester is started.
      # Make sure not file is defined twice as this can lead to unexpected behaviour.
    - paths:
        - /var/log/*.log

      # Type of the files. Based on this the way the file is read is decided.
      # The different types cannot be mixed in one prospector
      #
      # Possible options are:
      # * log: Reads every line of the log file (default)
      # * stdin: Reads the standard in
      input_type: log


    # Additional prospector

      # Configuration to use stdin input
    #- input_type: stdin
```

### PR DESCRIPTION
**Pull request by ruflin**
Closes #1033

### CODE REVIEW

**Comment by ruflin** [comment]
@dedemorton Can you have a look at this one? I hope I found all occurences of this pattern.

**Comment by dedemorton** [comment]
@ruflin LGTM. You might want to change the order of options in the reference topic about [configuration options](https://www.elastic.co/guide/en/beats/filebeat/current/configuration-filebeat-options.html) so that input_type comes first.

**Comment by ruflin** [comment]
@dedemorton I turned it around as suggested. And I realised that the change for this page was actually  missing.

**Comment by urso** [comment]
requires rebase

File: filebeat/docs/migration.asciidoc [hunk 1]
```diff
@@ -200,20 +199,20 @@ output:
...
+      certificate_authorities:
         - ./logstash-forwarder.crt
-        
+
       # Certificate for TLS client authentication
       certificate: ./logstash-forwarder.crt
 
       # Client Certificate Key
-      certificate_key: ./logstash-forwarder.key    
+      certificate_key: ./logstash-forwarder.key
```
Comment (urso) [inline filebeat/docs/migration.asciidoc]: btw. for SSL we don't need certificate and certificate_key. These two shall be used for client authentication.

Comment (ruflin) [inline filebeat/docs/migration.asciidoc]: @urso That part is coming from libbeat. Means we need to correct it there?

**Comment by ruflin** [comment]
@urso Rebased. I think no rebase was needed as the merge pull request button was still there. It seems like the interface of Github has changed. There is now also an update branch button (not sure what happens when clicked).

**Comment by monicasarbu** [comment]
I clicked the update branch button. Waiting for green "Merge pull request" button.

**Comment by tsg** [comment]
Update branch merges master into the feature branch. We should avoid it, as we decided to use rebases.

**Comment by urso** [comment]
hmmm... the button merges 'master' into current branch. Squashing commits + code cleanups will become painful if we click this button too often... let's avoid it.

**Comment by ruflin** [comment]
I will rebase again to get rid of the merge commit. We should not use the button.

File: filebeat/docs/migration.asciidoc (hunk 1 above)
Comment (urso) [inline filebeat/docs/migration.asciidoc]: yes

Comment (ruflin) [inline filebeat/docs/migration.asciidoc]: @urso are you going to take care of this?

**Pull request merged**